tests/e2e/pre_upgrade_test.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This test suite includes test cases to verify basic functionality
+// in most common configurations
+
+// +build runpreupgrade
+
+package e2e
+
+import (
+	"github.com/vmware/docker-volume-vsphere/tests/utils/dockercli"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/inputparams"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/misc"
+	"github.com/vmware/docker-volume-vsphere/tests/utils/verification"
+	. "gopkg.in/check.v1"
+)
+
+type PreUpgradeTestSuite struct {
+	config        *inputparams.TestConfig
+	esx           string
+	vm1           string
+	vm2           string
+	vm1Name       string
+	vm2Name       string
+	volName1      string
+	volName2      string
+	containerName string
+}
+
+func (s *PreUpgradeTestSuite) SetUpSuite(c *C) {
+	s.config = inputparams.GetTestConfig()
+	if s.config == nil {
+		c.Skip("Unable to retrieve test config, skipping basic tests")
+	}
+
+	s.esx = s.config.EsxHost
+	s.vm1 = s.config.DockerHosts[0]
+	s.vm2 = s.config.DockerHosts[1]
+	s.vm1Name = s.config.DockerHostNames[0]
+	s.vm2Name = s.config.DockerHostNames[1]
+}
+
+func (s *PreUpgradeTestSuite) SetUpTest(c *C) {
+	s.volName1 = "preUpgradeTestVol"
+	s.containerName = inputparams.GetUniqueContainerName(c.TestName())
+}
+
+var _ = Suite(&PreUpgradeTestSuite{})
+
+// Test steps:
+// 1. Create a volume
+// 2. Verify the volume is detached
+// 3. Attach the volume
+// 4. Verify volume status is attached
+// 5. Write some data to that volume
```

Comment:
write data, detach, attach, read - that involves the plugin and the esx service and verifies the mount/unmount workflows and checks that correct volume is bound to the VM each time.